### PR TITLE
extract slot-hashes crate

### PR DESCRIPTION
**Pull request by kevinheavey**
#### Problem
`solana_program::slot_hashes` blocks ripping out `solana_program::address_lookup_table` (which blocks ripping out `solana_program::message`)

#### Summary of Changes
- Move to a new crate and re-export in solana_program
- Make serde optional in the new crate

### CODE REVIEW

File: sdk/slot-hashes/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,112 @@
+//! A type to hold data for the [`SlotHashes` sysvar][sv].
+//!
+//! [sv]: https://docs.solanalabs.com/runtime/sysvars#slothashes
+//!
+//! The sysvar ID is declared in [`sysvar::slot_hashes`].
```
Comment (febo) [inline sdk/slot-hashes/src/lib.rs]: I think we need to update this comment. Is the sysvar to be declared in this crate, or will it be in a separate crate?

Comment (kevinheavey) [reply sdk/slot-hashes/src/lib.rs]: Yeah I think the Sysvar impl will need to stay with the other sysvar code i.e. not in this crate. Updated the comment

**Comment by febo** [comment]
Looks good - @yihau can you please accept the `solana-slot-hashes` crate?

**Comment by yihau** [comment]
✅

**Review by febo — APPROVED** [review]

**Pull request merged**